pkg/monitoring/metrics/virt-controller/vmsnapshot.go

```diff
@@ -0,0 +1,41 @@
+package virt_controller
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	snapshotv1 "kubevirt.io/api/snapshot/v1beta1"
+)
+
+var (
+	vmSnapshotMetrics = []operatormetrics.Metric{
+		VmSnapshotCreationTimestamp,
+	}
+
+	VmSnapshotCreationTimestamp = operatormetrics.NewGaugeVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_vmsnapshot_timestamp_seconds",
+			Help: "Unix timestamp of when the VM snapshot was created. Including the VM name, namespace, and snapshot name.",
+		},
+		[]string{"vm_name", "vm_namespace", "snapshot_name", "snapshot_namespace", "snapshot_uid"},
+	)
+)
+
+func NewVMSnapshotCreated(snapshot *snapshotv1.VirtualMachineSnapshot, content *snapshotv1.VirtualMachineSnapshotContent) {
```

Comment:
Is there a reason you use the content? you can get the vm name and namespace from:
snapshot.Spec.Source.Name/Namespace which also can't be nil.. you can add a validation the the snapshot.Spec.Source.Kind is "VirtualMachine" but I dont really believe we ever plan to have a different kind of source supported anyways

Reply:
@ShellyKa13  How do we correlate a snapshot to the specific VM?

Reply:
@ShellyKa13 there is no `snapshot.Spec.Source.Namespace` just `snapshot.Spec.Source.Name`.
so is the snapshot Namespace the same as the vm namespace? if yes then `snapshot.Namespace` will be enough

Reply:
Yeah right. you can use the snapshot ns.